byron/ledger/formal-spec/properties.tex

```diff
@@ -324,4 +326,295 @@ \subsubsection{PVBUMP Properties}
     \end{array}
   \right)
   $$
-\end{property}
\ No newline at end of file
+\end{property}
+
+
+\subsubsection{UPIEC Properties}
+\label{sec:upiec-properties}
+
+Property~\ref{UPIEC-no-change} states that if PVBUMP transitions into a state
+with the same protocol version, the UPIEC STS results in the same state it
+started in.
+
+\begin{property}[UPIEC no change on no protocol change]\label{UPIEC-no-change}
```

Comment:
Here and in the properties that follow we should universally quantify the variables in the start states.

Reply:
Haven't I done that already? The quantification is in the very first sentence of the `property` environment.

Reply:
As far as I understand, all the variables in the start states, e.g. `avs`, `rpus`, `raus`, etc, should be universally quantified:

![image](https://user-images.githubusercontent.com/175315/58829838-200d7c00-8649-11e9-85e3-9bbc974e5431.png)

Otherwise they are not bound to anything. And actually, the variables in the right hand side should be existentially quantified, since it seems we're stating:

For all `e`, `s`, such that the exists an `s'` such that

```
   e |- s ---> s'
```
then `P(e, s, s')`

Note that this different from the notation:

![image](https://user-images.githubusercontent.com/175315/58829952-6bc02580-8649-11e9-878d-fe3ba75f8c9f.png)

Since this is a shorthand for 

![image](https://user-images.githubusercontent.com/175315/58829988-7f6b8c00-8649-11e9-9776-80dacc1651ff.png)

Alternatively, we could state these properties like (here I'm again omitting the label (signal) for simplicity):

For all (valid transition) `e |- s ---> s'` such that `R(e, s, s')` we have that `P(e, s, s`)`.

I think I like the last one better, since it is more in line with the other properties.

A third alternative might be not to express these properties in the formal spec at all, since they're more or less stating the same thing as in the rules, and we keep them as sanity check for the implementation of the rules.

@nc6 any thoughts?

Reply:
My feeling is that we shouldn't express something like this in the formal spec at all. I would rather express only properties which have some "real" interpretation (e.g. are constraints on the system, rather than consequences of its implementation)

Reply:
I'm not sure what you mean with "real" interpretation, Nicholas. This is an actual STS that is part of a bigger system. Sure, the STS is a low-level STS, hence the properties for it are not too exciting. I'm expecting properties to get more interesting as we move to higher-level STSes.

@nc6, @dnadales and @uroboros , can we put notation, formatting and quantification issues aside for a moment and focus on what should properties be?

I'd say you have a different idea of what properties for this STS should be and I'll be happy to go with your proposals; I'm trying to see how to put your proposals together, but it's not really clear.

I'll try to summarise here what you'd guys like to see and let me know if that's ok:

- drop properties 1-3 as they're trivial
- take only property 4 as it is the only one expressing a constraint on the system; reword its prose and motivation.

Does that sound good?

Reply:
>  I'm expecting properties to get more interesting as we move to higher-level STSes.

As a side note, maybe we should have begun by adding those high level properties to the ones we already have. 

I think that "real interpretation" in this context means that the property can be traced back to some (functional) high level requirement like:

- updates eventually get though
- no minority can prevent updates from happening
- money supply remains constant
- etc 

> @nc6, @dnadales and @uroboros , can we put notation, formatting and quantification issues aside for a moment and focus on what should properties be?

If we follow @nc6's advice (which I think we should) we can simply remove these low level properties from the formal spec.

> take only property 4 as it is the only one expressing a constraint on the system; reword its prose and motivation.

Maybe we can have a call about this with @nc6 to reach an agreement?